src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/ApiWebSecurityConfigurerAdapter.java

```diff
@@ -92,42 +93,12 @@ protected void configure(HttpSecurity http) throws Exception {
             .formLogin()
                 .disable();
     }
-    /**
-     * Checks whether session cookie auth, or some other (api key) auth was used.
-     * Purely based on existence of JSESSIONID cookie.
-     */
-    private static boolean isSessionCookieAuthenticated(HttpServletRequest request) {
-        return WebUtils.getCookie(request, "JSESSIONID") != null;
-    }
 
     /**
-     * CookieCsrfTokenRepository (wrapper) which does not send unneeded CSRF cookies if we're
-     * using api key auth
+     * Check if an alive session exists
      */
-    private static class ApiKeyAuthAwareCookieCsrfTokenRepository implements CsrfTokenRepository {
-
-        // final class, cannot be extended
-        private CookieCsrfTokenRepository cookieCsrfTokenRepository;
-
-        ApiKeyAuthAwareCookieCsrfTokenRepository() {
-            this.cookieCsrfTokenRepository = CookieCsrfTokenRepository.withHttpOnlyFalse();
-        }
-        @Override
-        public CsrfToken generateToken(HttpServletRequest request) {
-            return cookieCsrfTokenRepository.generateToken(request);
-        }
-
-        @Override
-        public void saveToken(CsrfToken token, HttpServletRequest request, HttpServletResponse response) {
-            if (isSessionCookieAuthenticated(request)) {
-                cookieCsrfTokenRepository.saveToken(token, request, response);
-            }
-        }
-
-        @Override
-        public CsrfToken loadToken(HttpServletRequest request) {
-            return cookieCsrfTokenRepository.loadToken(request);
-        }
+    private static boolean isSessionCookieAuthenticated(HttpServletRequest request) {
+        return request.getSession(false) != null;
```

Comment:
I don't remember why I implemented the original `isSessionCookieAuthenticated` as `WebUtils.getCookie(request, "JSESSIONID") != null;` instead. It is a little bit different, and I think you said original one did not work in some situations properly, which seems curious. 

I had some thoughts about renaming the method since it is not 100% about cookies (and session id could be in path params) but maybe it is more clear the way it already is.

Reply:
Renamed the method